pkg/ivd/ivd_protected_entity.go

```diff
@@ -295,13 +295,20 @@ func (this IVDProtectedEntity) Snapshot(ctx context.Context, params map[string]m
 	retrieveSnapDetailsErr := 0
 	err := wait.PollImmediate(retryInterval, time.Hour, func() (bool, error) {
 		this.logger.Infof("Retrying CreateSnapshot on IVD Protected Entity, %v, for one hour at the maximum, Current retry count: %d", this.GetID().String(), retryCount)
+		retryCount++
 		var vslmTask *vslm.Task
 		vslmTask, err := this.ipetm.vslmManager.CreateSnapshot(ctx, NewVimIDFromPEID(this.GetID()), "AstrolabeSnapshot")
 		if err != nil {
+			if soap.IsSoapFault(err) {
+				fault := soap.ToSoapFault(err).Detail.Fault
+				if _, ok := fault.(vim.NotFound); ok {
```

Comment:
Could this "NotFound" be caused by something else other than FCD not found?
Can you add a comment here?